### PR TITLE
Add codecov support

### PR DESCRIPTION
**Pull request by aeschylus**

### CODE REVIEW

**Comment by jkeck** [comment]
Is it possible to hook this sort of check up w/ a GitHub service so that it reports it as a build status (perhaps using CodeClimate or something along those lines)?

This looks good to me, so I'm 👍 to merging and figure that out later, but might be nice if it showed as a failed build step if we reduce status. (or maybe this fails the build?)

**Comment by mejackreed** [comment]
Nice! Should we also think about ignoring the directory where the coverage gets sent so someone locally doesn't commit?

**Comment by aeschylus** [comment]
This is not ready to merge, sorry for lack of label. The connection with the service is not complete. I will remove the coverage folder as well.

I've asked to add the codecov App to our top-level org, which is how their integration now works. I don't see a settings tab under the organisation, so someone should give me permissions to sul-dlss to finish that.

Also, apparently we need to revisit travis and other integrations as well:
screen shot 2018-09-19 at 10 07 26 am

File: package-lock.json [hunk 1]
```diff
@@ -387,6 +387,12 @@
         "sprintf-js": "~1.0.2"
       }
     },
+    "argv": {
```
Comment (mejackreed) [inline package-lock.json]: Since no packages are being added or removed, should we be diffing this file?

Comment (aeschylus) [reply package-lock.json]: In this case the `codecov` package is being added, but I agree we that package.json is still causing trouble occasionally. What would you recommend? `git rm`, or just remove it from the commit?

Comment (mejackreed) [reply package-lock.json]: Ahh .. I missed that totally. Thanks for clarifying. I think its good / or ok since we are changing a package.

**Pull request merged**